### PR TITLE
Support SQL Server

### PR DESCRIPTION
**Pull request by junpataleta**
Since Moodle officially supports SQL Server and with docker making running SQL Server easier even on Linux nowadays, I think it would be nice if MDK would also be able to support creation/installation of Moodle instances running with the SQL Server database.

### CODE REVIEW

File: mdk/db.py [hunk 1]
```diff
@@ -69,6 +69,22 @@ def __init__(self, engine, options):
...
+            password = str(options['passwd'])
+            connectionstr = "DRIVER={ODBC Driver 13 for SQL Server};SERVER=%s;PORT=%d;UID=%s;PWD=%s" \
+                            % (host, port, user, password)
+            self.conn = pyodbc.connect(connectionstr)
+            self.conn.autocommit = True
+
+            try:
+                self.cur = self.conn.cursor()
+            except:
+                raise Exception('Connexion failed! Make sure the database \'%s\' exists.' % str(options['user']))
```
Comment (FMCorz) [inline mdk/db.py]: Is this error message relevant, or a copy/paste from Postegres left behind? With postgres, the connection always fails if a database of the same name of the user does not also exist.

File: requirements.txt [hunk 2]
```diff
@@ -4,3 +4,4 @@ MySQL-python>=1.2.3
 psycopg2>=2.4.5
 requests>=2.3.0
 watchdog>=0.8.0
+pyodbc>=4.0.21
```
Comment (FMCorz) [inline requirements.txt]: Could you add the Linux dependencies to the README file if relevant? It seems that additional packages need installing for pyodbc to work...

Ref: https://github.com/mkleehammer/pyodbc/wiki/Install

**Comment by junpataleta** [comment]
Thanks, Fred.
1. Yep, copy pasted from PostgreSQL and totally missed that. (sorry!)
2. Good catch. I added the unixodbc-dev dependency in the README file.

Let me know if this fine or if I should just squash the commits.

Thanks!

**Comment by FMCorz** [comment]
Thanks!

**Pull request merged**